model/token.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package model
+
+import (
+	"encoding/json"
+	"time"
+
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+	"github.com/mendersoftware/go-lib-micro/mongo/oid"
+
+	"github.com/mendersoftware/useradm/jwt"
+)
+
+type TokenRequest struct {
+	Name      *string `json:"name"`
+	ExpiresIn int64   `json:"expires_in"`
+}
+
+const defaultTokenMaxExpiration = 31536000
+
+func (tr TokenRequest) Validate(maxExpiration int) error {
+	if maxExpiration <= 0 {
+		maxExpiration = defaultTokenMaxExpiration
+	}
+	return validation.ValidateStruct(&tr,
+		validation.Field(&tr.Name, validation.Required),
```

Comment:
could we have an upper limit on the name length?

Reply:
done